File: backend/app/models/mesa.py
# Importaciones necesarias para el modelo
from sqlalchemy import Column, Integer, ForeignKey
from sqlalchemy.orm import relationship
from app.db.base_class import Base

class Mesa(Base):
    """
    Modelo que representa una mesa de juego en un campeonato.
    Cada mesa representa un enfrentamiento entre dos parejas en una partida específica.
    
    Attributes:
        id (int): Identificador único de la mesa
        numero (int): Número asignado a la mesa en el campeonato
        campeonato_id (int): ID del campeonato al que pertenece la mesa
        partida (int): Número de la partida en la que se juega
        pareja1_id (int): ID de la primera pareja asignada a la mesa
        pareja2_id (int): ID de la segunda pareja asignada a la mesa (puede ser null)
    """
    # Nombre de la tabla en la base de datos
    __tablename__ = "mesas"

    # Columnas de la tabla
    id = Column(Integer, primary_key=True, index=True)                     # ID único de la mesa
    numero = Column(Integer, nullable=False)                               # Número de mesa en el campeonato
    campeonato_id = Column(Integer, ForeignKey("campeonatos.id"))         # Referencia al campeonato
    partida = Column(Integer, nullable=False)                             # Número de partida
    pareja1_id = Column(Integer, ForeignKey("parejas.id"))                # Referencia a la primera pareja
    pareja2_id = Column(Integer, ForeignKey("parejas.id"), nullable=True) # Referencia a la segunda pareja (opcional)

    # Relaciones con otros modelos
    campeonato = relationship("Campeonato", back_populates="mesas")       # Relación con el campeonato
    pareja1 = relationship(
        "Pareja", 
        foreign_keys=[pareja1_id], 
        back_populates="mesas_como_pareja1"
    )  # Relación con la primera pareja
    pareja2 = relationship(
        "Pareja", 
        foreign_keys=[pareja2_id], 
        back_populates="mesas_como_pareja2"
    )  # Relación con la segunda pareja
    resultados = relationship("Resultado", back_populates="mesa")         # Relación con los resultados

    def to_dict(self):
        """
        Convierte el objeto Mesa a un diccionario.
        Incluye información básica de la mesa y los nombres de las parejas.
        
        Returns:
            dict: Diccionario con los atributos de la mesa y nombres de parejas
        
        Note:
            Los nombres de las parejas solo se incluyen si las parejas existen
        """
        return {
            "id": self.id,
            "numero": self.numero,
            "campeonato_id": self.campeonato_id,
            "partida": self.partida,
            "pareja1_id": self.pareja1_id,
            "pareja2_id": self.pareja2_id,
            "pareja1_nombre": self.pareja1.nombre if self.pareja1 else None,
            "pareja2_nombre": self.pareja2.nombre if self.pareja2 else None
        }

File: frontend/src/types/mesa.ts
/**
 * Definición de tipos y interfaces relacionadas con las parejas y mesas del campeonato
 */

/**
 * Interface que define la estructura de una Pareja en el campeonato
 * @interface Pareja
 * @property {number} id - Identificador único de la pareja
 * @property {string} nombre - Nombre de la pareja
 * @property {string} [club] - Club al que pertenece la pareja (opcional)
 * @property {boolean} activa - Estado de la pareja (activa/inactiva)
 * @property {number} [numero] - Número asignado a la pareja en el campeonato (opcional)
 * @property {number} campeonato_id - ID del campeonato al que pertenece la pareja
 */
export interface Pareja {
  id: number
  nombre: string
  club?: string
  activa: boolean
  numero?: number
  campeonato_id: number
}

/**
 * Interface que define la estructura básica de una Mesa de juego
 * @interface Mesa
 * @property {number} id - Identificador único de la mesa
 * @property {number} numero - Número de la mesa en el campeonato
 * @property {number} campeonato_id - ID del campeonato al que pertenece la mesa
 * @property {number} partida - Número de la partida actual
 * @property {number} pareja1_id - ID de la primera pareja asignada
 * @property {number} [pareja2_id] - ID de la segunda pareja asignada (opcional)
 * @property {boolean} [tieneResultado] - Indica si la mesa tiene resultado registrado
 * @property {Pareja} [pareja1] - Objeto con los datos de la primera pareja
 * @property {Pareja} [pareja2] - Objeto con los datos de la segunda pareja
 */
export interface Mesa {
  id: number
  numero: number
  campeonato_id: number
  partida: number
  pareja1_id: number
  pareja2_id?: number
  tieneResultado?: boolean
  pareja1?: Pareja
  pareja2?: Pareja
}

/**
 * Interface que define la estructura base de un Resultado
 * @interface ResultadoBase
 * @property {number} [id] - Identificador único del resultado
 * @property {number} campeonato_id - ID del campeonato
 * @property {number} partida - Número de la partida
 * @property {number} mesa_id - ID de la mesa donde se jugó
 * @property {number} id_pareja - ID de la pareja a la que pertenece el resultado
 * @property {number} RP - Puntos de resultado
 * @property {number} PG - Partidas ganadas
 * @property {number} PP - Partidas perdidas
 * @property {string} GB - Diferencial de juegos (Game Balance)
 */
export interface ResultadoBase {
  id?: number
  campeonato_id: number
  partida: number
  mesa_id: number
  id_pareja: number
  RP: number
  PG: number
  PP: number
  GB: string
}

/**
 * Interface que define la estructura de respuesta de un Resultado
 * @interface ResultadoResponse
 * @property {ResultadoBase} pareja1 - Resultado de la primera pareja
 * @property {ResultadoBase} [pareja2] - Resultado de la segunda pareja (opcional)
 */
export interface ResultadoResponse {
  pareja1: ResultadoBase
  pareja2?: ResultadoBase
}

/**
 * Interface que extiende Mesa para incluir los resultados
 * @interface MesaConResultados
 * @extends Mesa
 * @property {ResultadoResponse} [resultados] - Objeto con los resultados de ambas parejas
 * @property {boolean} tieneResultados - Indica si la mesa tiene resultados registrados
 */
export interface MesaConResultados extends Mesa {
  resultados?: ResultadoResponse
  tieneResultados: boolean
} 